lib/resqued/exec_on_hup.rb

```diff
@@ -0,0 +1,42 @@
+require "yaml"
+
+module Resqued
+  class ExecOnHUP
+    # Public: Replace the current master process with a new one, while preserving state.
+    def self.exec!(state)
+      exec Resqued::START_CTX['$0'], "--replace", store_state(state), exec_opts(state)
+    end
+
+    # Internal: Returns exec options for each open socket in 'state'.
+    def self.exec_opts(state)
+      exec_opts = {}
+      state.sockets.each do |sock|
+        exec_opts[sock.to_i] = sock
+      end
+      if pwd = Resqued::START_CTX['pwd']
+        exec_opts[:chdir] = pwd
+      end
+      return exec_opts
+    end
+
+    # Internal: Write out current state to a file, so that a new master can pick up from where we left off.
+    def self.store_state(state)
+      data = {version: Resqued::VERSION}
```

Comment:
I think our github/github rubucop config requires spaces around this, dunno if we care about that here?